docs/limitations.rst

```diff
@@ -92,100 +93,161 @@ the "sync-barrier-sync" construct needs to be used.
+To run correctly on UnifyFS, an application must make
+all ``MPI_File_sync()`` calls required by the MPI standard.
 
-    This option will soon be available in the `HDF5 develop branch`_ as well as
-    in the next HDF5 release.
+**Cost:** It may be labor intensive to identify and correct all places
+within an application where file synchronization calls are required.
+The :doc:`VerifyIO <verifyio>` tool can assist with this effort.
 
-**Cost:** Can cause a significant decrease in write performance as the amount of
-file sync operations performed will likely be more than necessary. Similar to,
-but potentially more efficient than, the ``WRITE_SYNC`` workaround as less
-overall file syncs may be performed in comparision, but still likely more than
-needed.
+.. TODO: Mention use/need of ``romio_visibility_immediate`` hint once available.
+.. https://github.com/pmodels/mpich/issues/5902
 
-ROMIO Driver Hint
-"""""""""""""""""
+---------------------------
+ROMIO Limitations
+---------------------------
 
-A ROMIO provided hint that will cause the ROMIO driver (in a supported MPI
-library) to add an ``MPI_Barrier()`` call and an additional ``MPI_File_sync()``
-call after each already existing ``MPI_File_sync()`` call within the
-application. In other words, this hint converts each existing
-``MPI_File_sync()`` call into the "sync-barrier-sync" construct. Enable the
-``romio_synchronizing_flush`` hint to use this workaround.
+Synchronization
+***************
 
-**Cost:** Potentially more efficient that the ``WRITE_SYNC`` and HDF5
-``FILE_SYNC`` workarounds as this will cause the application to use the
-synchronization construct required by UnifyFS everywhere the application already
-intends them to occur (i.e., whenever there is already an ``MPI_File_sync()``).
-However, if (1) any existing ``MPI_File_sync()`` calls are only meant to make
-data visible to the other processes (rather than to avoid potential conflicts)
-or (2) the application contains a mix of lone ``MPI_File_sync()`` calls along
-with the "sync-barrier-sync" construct, then this approach will result in more
-syncs than necessary.
+In ROMIO, ``MPI_File_sync()`` calls ``fsync()``
+and ``MPI_File_close()`` calls ``close()``,
+each of which flush information about newly
+written data to the UnifyFS servers.
+When using ROMIO, an application with appropriate
+"sync-barrier-sync" constructs as required by MPI
+should run correctly on UnifyFS.
 
-----------
+ROMIO Driver Hint
+"""""""""""""""""
 
-.. _file-lock-label:
+Although ``MPI_File_sync()`` is an MPI collective,
+it is not required to be synchronizing.
+One can configure ROMIO such that ``MPI_File_sync()``
+is also a synchronizing collective.
+This configuration can be useful to applications that
+only call ``MPI_File_sync()`` once rather than a full
+sync-barrier-sync construct.
+To enable this behavior, one can set the ROMIO hint
+``romio_synchronizing_flush true`` though an ``MPI_Info``
+object or in a ``ROMIO_HINTS`` file.
```

Comment:
We should list the ROMIO/MPICH version where this hint first appeared.

Reply:
This appears to have been merged at the end of 2021 (https://github.com/pmodels/mpich/pull/5593)
and became available in v4.0
https://github.com/pmodels/mpich/blob/9d9d208011b7714daf9ef680f015bb18ff3aa5b6/src/mpi/romio/adio/include/adioi.h#L50

Reply:
Thanks @CamStan in tracking down this commits and corresponding versions.  I just posted those notes as reminders for myself :-)